### PR TITLE
[tests/CI] Test module import

### PR DESCRIPTION
**Pull request by viktorasl**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
In the future it would be best to avoid cases like (https://github.com/fastlane/fastlane/issues/14661).

### Description
The idea is to execute `test_modules/run_tests.sh` file which execute all dummy ruby import tests scripts. I'm quite out of ideas now how to produce "test cases" from shell script.

Any input is welcomed! Or I can just close it and open an issue instead.

Still WIP. Made PR just to be able to test CI.

### CODE REVIEW

**Comment by viktorasl** [comment]
Managed to produce failing CI step:
https://circleci.com/gh/fastlane/fastlane/31759?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by janpio** [comment]
Awesome!

Can you add this as its own build step in the Circle config? 
What would be an appropriate name for these checks? 
What other checks should this be doing?

**Comment by viktorasl** [comment]
Created a CI step named `Execute module load up tests` without used cache (probably someone should elaborate on that).
So this PR now has MVP structure of how I imagine these tests.
Btw I'm not a ruby guy (nor bash scripting guy) so I would appreciate if someone with ruby experience would take a look if this solution is viable.

File: .circleci/config.yml [hunk 1]
```diff
@@ -144,6 +144,23 @@ jobs:
...
+    docker:
+      - image: circleci/ruby:2.3
+    shell: /bin/bash --login -eo pipefail
+    steps:
+      - checkout
+
+      - run:
+          name: Setup Build
+          command: |
+            gem install $(echo $(gem build fastlane.gemspec) | sed 's/.*File: //')
```
Comment (janpio) [inline .circleci/config.yml]: What does this do?

File: test_modules/run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+# running tests
+TEST_FILES=(
+  "test_modules/test_spaceship.rb"
+  "test_modules/test_scan.rb"
+)
+EXIT_CODE=0
+for TEST_FILE in ${TEST_FILES[*]} 
+do
+  ruby $TEST_FILE
```
Comment (janpio) [inline test_modules/run_tests.sh]: Per https://stackoverflow.com/a/8795650/252627 you should be able to just do `ruby -e "require 'spaceship'"` here to achieve the same result - and avoid needing those files, but that doesn't work for me locally :/

Comment (viktorasl) [reply test_modules/run_tests.sh]: omg! I searched for something like this and didn't manage to find 😫

File: .circleci/config.yml (hunk 1 above)
Comment (viktorasl) [reply .circleci/config.yml]: It builds gem from gemspec and installs it right away

Comment (janpio) [reply .circleci/config.yml]: The wonders of shell/bash stuff...

(Might want to add `Build gem from fastlane.gemspec and install locally` or similar as a comment above)

**Comment by viktorasl** [comment]
Tweaked a bit 👍

**Comment by janpio** [comment]
Interesting that `ruby -e` actually works for you and on CI, as it didn't locally on my machine. But whatever 🤷‍♀ 

So this proof of concept definitely works. Now the question is what else should be included here. Maybe all the toplevel modules? (basically the folders in the root of the repo)

Hey Ruby people, what do you think?

**Comment by snatchev** [comment]
@viktorasl looks like you're test works and caught the failure :)

Could you rebase off of master to get this to pass.

I don't quite understand the changes in .circleci.yaml but otherwise LGTM.

**Comment by janpio** [comment]
> I don't quite understand the changes in .circleci.yaml but otherwise LGTM

I can approve those changes, it adds another "job" that builds and installs the local gem and then runs the script - so the circle part looks good.

**Comment by viktorasl** [comment]
Yep, it worked as expected after merge of master (I did not rebase on master, but unless you require I can do that).
If you approve I'll add outstanding modules in the test list.

**Comment by janpio** [comment]
Go for it!

**Comment by viktorasl** [comment]
Done. Works well, but only tests top level module load up (not testing i.e `pem/manager` and similar)

**Comment by janpio** [comment]
It does. 

Let's get this merged then, so please remove the `WIP` from the title @viktorasl.

**Review by janpio — APPROVED** [review]

**Comment by viktorasl** [comment]
Ready to be merged. I wanted to style (add colors in terminal for success / failure) tests log but I'll leave it for the future. I have to gain more experience in shell scripting.

**Pull request merged**